File: algorithms/go/fibonacci-recursive.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

const MOD = 1_000_000_007

func fib(n int) int {
	if n <= 1 { return n }
	return (fib(n-1) + fib(n-2)) % MOD
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscan(reader, &n)

	start  := time.Now()
	result := fib(n)
	ms     := float64(time.Since(start).Nanoseconds()) / 1e6

	fmt.Fprintln(writer, result)
	fmt.Fprintf(os.Stderr, "%.3f\n", ms)
}

File: algorithms/c/fibonacci-recursive.c
#include <stdio.h>
#include <time.h>

#define MOD 1000000007

long long fib(int n) {
    if (n <= 1) return n;
    return (fib(n-1) + fib(n-2)) % MOD;
}

int main() {
    int n;
    scanf("%d", &n);

    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);
    long long result = fib(n);
    clock_gettime(CLOCK_MONOTONIC, &end);

    double ms = (end.tv_sec  - start.tv_sec)  * 1000.0
              + (end.tv_nsec - start.tv_nsec) / 1e6;

    printf("%lld\n", result);
    fprintf(stderr, "%.3f\n", ms);
    return 0;
}